pkg/controller/jobs/leaderworkerset/leaderworkerset_reconciler.go

```diff
@@ -0,0 +1,227 @@
+	}
+
+	log := ctrl.LoggerFrom(ctx).WithValues("leaderworkerset", klog.KObj(lws))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling LeaderWorkerSet")
+
+	err = r.createPrebuildWorkloadsIfNotExist(ctx, lws)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) createPrebuildWorkloadsIfNotExist(ctx context.Context, lws *leaderworkersetv1.LeaderWorkerSet) error {
+	replicas := ptr.Deref(lws.Spec.Replicas, 1)
+	for i := int32(0); i < replicas; i++ {
+		if err := r.createPrebuildWorkloadIfNotExist(ctx, lws, i); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) createPrebuildWorkloadIfNotExist(ctx context.Context, lws *leaderworkersetv1.LeaderWorkerSet, index int32) error {
+	wl := &kueue.Workload{}
+	err := r.client.Get(ctx, client.ObjectKey{Name: GetWorkloadName(lws.UID, lws.Name, fmt.Sprint(index)), Namespace: lws.Namespace}, wl)
+	// Ignore if the Workload already exists or an error occurs.
+	if err == nil || client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if err := r.createPrebuildWorkload(ctx, lws, index); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) createPrebuildWorkload(ctx context.Context, lws *leaderworkersetv1.LeaderWorkerSet, index int32) error {
+	createdWorkload := r.constructWorkload(lws, index)
+	err := r.client.Create(ctx, createdWorkload)
+	if err != nil {
+		return err
+	}
+	r.record.Eventf(
+		lws, corev1.EventTypeNormal, jobframework.ReasonCreatedWorkload,
+		"Created Workload: %v", workload.Key(createdWorkload),
+	)
+	return nil
+}
+
+func (r *Reconciler) constructWorkload(lws *leaderworkersetv1.LeaderWorkerSet, index int32) *kueue.Workload {
+	wl := &kueue.Workload{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        GetWorkloadName(lws.UID, lws.Name, fmt.Sprint(index)),
+			Namespace:   lws.Namespace,
+			Finalizers:  []string{kueue.ResourceInUseFinalizerName},
+			Annotations: admissioncheck.FilterProvReqAnnotations(lws.GetAnnotations()),
+		},
+		Spec: kueue.WorkloadSpec{
+			QueueName:                   jobframework.QueueNameForObject(lws),
+			PodSets:                     r.podSets(lws),
+			MaximumExecutionTimeSeconds: jobframework.MaximumExecutionTimeSecondsForObject(lws),
+		},
+	}
+
+	wl.Annotations[podcontroller.IsGroupWorkloadAnnotationKey] = podcontroller.IsGroupWorkloadAnnotationValue
+
+	return wl
+}
+
+func (r *Reconciler) podSets(lws *leaderworkersetv1.LeaderWorkerSet) []kueue.PodSet {
+	podSets := make([]kueue.PodSet, 0, 2)
+
+	if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
+		podSets = append(podSets, kueue.PodSet{
+			Name:  leaderPodSetName,
+			Count: 1,
+			Template: corev1.PodTemplateSpec{
+				Spec: *lws.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec.DeepCopy(),
+			},
+			TopologyRequest: jobframework.PodSetTopologyRequest(
+				&lws.Spec.LeaderWorkerTemplate.LeaderTemplate.ObjectMeta,
+				// TODO: ???
+				ptr.To(kueuealpha.PodGroupPodIndexLabel),
```

Comment:
IIUC this is for TAS rank-based ordering, but does it work? IIUC STS will set indexes for workers, so we should use this label. The rank-based ordering for leaders does not matter, because there is only one leader. Please make sure there is an e2e test added ( could be a follow up). 

Either support TAS in this PR with the label, or leave the TAS support with e2e test for a follow up.